libc/functions/i/ioctl.part-impl.md

```diff
@@ -0,0 +1,81 @@
+# Synopsis 
+`#include <ioctl.h>`</br>
+
+` int ioctl(int fildes, int request, ... /* arg */);`</br>
+
+## Status
+Partially implemented
+## Conformance
+IEEE Std 1003.1-2017
+## Description
+
+
+The `ioctl()` function shall perform a variety of control functions on devices. For _fildes_ not referring to devices, the functions performed by this call are unspecified.
+
+The _fildes_ argument is an open file descriptor that refers to a device.
+
+The _request_ argument selects the control function to be performed and shall depend on the device being addressed.
+
+The _arg_ argument represents additional information that is needed by this specific device to perform the requested function. The type of _arg_ depends upon the particular control request, but it shall be either an integer or a pointer to a device-specific data structure.
+
+In order to define driver request one should use one of the macros presented below, which are available in the `ioctl.h` header file:
+
+    _IOC(inout,group,num,len) - macro which return every possible request
+
+_inout_ - use these defines: IOC_VOID (no data), IOC_OUT (read), IOC_IN (write), IOC_INOUT (read/write)
+_group_ - describe group type (collecting the ioctls in groups for a common purpose or a common driver), 8-bit value
+_num_ - specifie serial number of request, 8 -bit value
+_len_ - how big is size of data, 14-bit value
+
+    _IOV(g,n) - macro which return no data transmission request
+    _IOR(g,n,t) - macro which return read request
+    _IOW(g,n,t) - macro which return write request
+    _IOWR(g,n,t) - macro which return read/write request
```

Comment:
We can firstly list all the possible macros like this and then describe arguments in a similar way:
```suggestion
* `_IOC(inout,group,num,len)` - macro which return every possible request
* `_IOV(g,n)` - macro which returns request without data transmission,
* _IOR(g,n,t) - macro which returns read request,
* _IOW(g,n,t) - macro which returns write request,
* _IOWR(g,n,t) - macro which returns read/write request,

Arguments have been described below:
* _inout_ - use these defines: `IOC_VOID` (no data), IOC_OUT (read), IOC_IN (write), IOC_INOUT (read/write)
* _group_ - describe group type (collecting the ioctls in groups for a common purpose or a common driver), 8-bit value
* _num_ - specifie serial number of request, 8 -bit value
* _len_ - how big is size of data, 14-bit value
```

 - It's always good to use `*` or `-` when you list sth.
 
  - Also remember about highlighting `IOC_VOID` and so on. Suggestion is not fully completed.
  
  - macro which return -> macro which returns

Reply:
As we discussed you want to indicate what should be passed as _group_ argument (typically a uppercase letter, for example 'S' for sockets <- sth like this)